flycheck.el

```diff
@@ -2787,6 +2842,11 @@ buffer."
     (when flycheck-mode
       ;; The buffer was changed, thus clear the idle timer
       (flycheck-clear-idle-change-timer)
+      ;; If we can guarantee that this timer is actually going to
+      ;; result in a syntax check, we can cancel any other
+      ;; pre-existing timers that would cause syntax checks.
+      (when (memq 'idle-change flycheck-check-syntax-automatically)
+        (flycheck-clear-idle-buffer-switch-timer))
```

Comment:
Instead of having two separate timers that cancel each other, can't we just keep one timer variable?  That way, we would just have to check whether the timer is set, cancel it if it is, and set its new value.  We wouldn't have to peek into `flycheck-check-syntax-automatically`.

Maybe name it `flycheck-idle-trigger-timer`.

Reply:
@fmdkdd Are you okay with the user not being able to customize the delay for checking after an edit separately from the delay for checking after a buffer switch?

Reply:
We keep the two delays, but we use only one `timer` variable for the results of `run-at-time`.  This is just a simpler way to ensure only one "delayed action" can exist at a time.  Do you see what I mean?

Reply:
I think that could work. I'll look into it.